app/services/katello/pulp3/repository.rb

```diff
@@ -218,6 +218,33 @@ def self.list(smart_proxy, args)
         ::Katello::Pulp3::Repository.new(nil, smart_proxy).list(args)
       end
 
+      def self.repository_versions(smart_proxy)
+        current_repos = list(smart_proxy, {})
```

Comment:
i don't think this would work across pages, would it?

Reply:
You know, you are right, it would not work across pages. Good point.

Reply:
Looks like its still not being paged?